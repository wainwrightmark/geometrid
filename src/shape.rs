use crate::{vector::Vector, location::HasLocation, flippable::Flippable, rotatable::Rotatable};


/// A general shape made of tiles. 
/// These are represented by vectors
pub trait Shape : Flippable + Rotatable + HasLocation + IntoIterator<Item = Self::Vector>{
    type Vector: Vector;

    
}