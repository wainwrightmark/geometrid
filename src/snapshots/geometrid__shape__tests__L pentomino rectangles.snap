---
source: src/shape.rs
expression: rectangles
---
[
    Rectangle {
        x: 0,
        y: 1,
        width: 2,
        height: 1,
    },
    Rectangle {
        x: 0,
        y: -2,
        width: 1,
        height: 3,
    },
]
